Update MooseServer autocompletion lookups to use warehouses

brandonlangley opened the pull request:
refs #22766

brandonlangley committed:
Include objects from warehouses in MooseServer completion lookups

refs #22766

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/3440443/), step Docs: sync website on c0372fe wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/32067/site)

This comment will be updated on new commits.

brandonlangley commented:
@loganharbour @dschwen -

This updates the MooseServer autocompletion to use warehouses in support of action generated objects including:

- `MaterialName` types using the problem `MaterialWarehouse` objects
- `MaterialPropertyName` types using the `MaterialPropertyRegistry` names
- `FunctionName` types using the problem `FunctionWarehouse` objects
- `OutputName` types using the `OutputWarehouse` names and reserved words
- `UserObjectName` types using `TheWarehouse` parameterized attribute queries
- `NonlinearVariableName` types using the `NonlinearSystem` solver variables
- `AuxVariableName` types using the `AuxiliarySystem` equation variables
- `VariableName` types using `NonlinearSystem` and `AuxiliarySystem` names

The autocompletion is now using the `problem` from the `actionWarehouse` in order to access these various objects.

So I also updated the server logic so it continues to build the app if the parsing fails and run app if the building fails.

This lets the `problem` still be available for warehouse based autocompletion and plotting even if there was an error.

It previously did not build the app if parsing failed, and it deleted the app if building or running failed for any reason.

Without this change, warehouses could not be used to complete valueless parameters that cause a parse error like:

```
material =
```

Warehouses were also unavailable when something like duplicate parameters in an unrelated block caused an error.

I also added unit tests for all the above warehouse cases. Could one of you review this and merge it if you approve?

brandonlangley commented:
@loganharbour -

> I already looked at it. There's a comment by me above your comment.

I replied to that comment last night about why the `if (_parser.queryCommandLineRoot())` is needed. Do you see it?
